File: src/main/python/me/dongqinglin/UdpServer.py
import socket

localIP = "127.0.0.1"
localPort = 8081
bufferSize = 1024

def main():
    udpServer = socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM)
    udpServer.bind((localIP, localPort))
    print("UDP server starting at {}:{}".format(localIP, localPort))
    while(True):
        print("Waiting for a  UDP  packet...");
        packet = udpServer.recvfrom(bufferSize)
        message = packet[0]
        address = packet[1]
        print(message)
    pass

main()

File: src/main/java/me/dongqinglin/UdpServer.java
package me.dongqinglin;

import java.io.IOException;
import java.net.*;

public class UdpServer {
    public static String LOCAL_HOST_IP = "localhost";
    public static int PORT = 8080;


    public static void main(String[] args) throws IOException {
        DatagramSocket entryPoint = new DatagramSocket(PORT, InetAddress.getByName(LOCAL_HOST_IP));
        System.out.println("Created UDP  server socket at " + LOCAL_HOST_IP + ":" + PORT);
        while(true) {
            System.out.println("Waiting for a  UDP  packet...");
            DatagramPacket datagramPacket = new DatagramPacket(new byte[1024], 1024);
            entryPoint.receive(datagramPacket);
            displayPacketDetails(datagramPacket);
        }
    }

    public static void displayPacketDetails(DatagramPacket packet) {
        byte[] data = packet.getData();
        String msg = new String(data, packet.getOffset(), packet.getLength());
        System.out.println(msg);
    }

}
